packages/emitter-framework/src/python/components/class-declaration.tsx

```diff
@@ -0,0 +1,38 @@
+import { Children, Declaration, DeclarationProps, Indent, mapJoin, Scope } from "@alloy-js/core";
+import { usePythonNamePolicy } from "../name-policy.js";
+import { Decorator, DecoratorProps } from "./decorator.jsx";
+import { joinChildren } from "./util.js";
+
+/**
+ * Represents the properties for a class declaration.
+ */
+export interface ClassDeclarationProps extends DeclarationProps {
+  decorators?: DecoratorProps[];
+  extends?: Children;
+}
+
+export function ClassDeclaration(props: ClassDeclarationProps) {
+  if (!props.name) {
```

Comment:
why not rely on the interface type?

Reply:
This was lifted from the alloy version of this component in which case name needed to be required. For the TSP variant, it does not need to be, since we can borrow the name from the type. Updated.